java/kotlin-extractor2/src/main/kotlin/entities/Types.kt

```diff
@@ -1,31 +1,30 @@
 package com.github.codeql
 
+import com.github.codeql.entities.getTypeParameterLabel
 import org.jetbrains.kotlin.analysis.api.KaSession
+import org.jetbrains.kotlin.analysis.api.components.DefaultTypeClassIds
 import org.jetbrains.kotlin.analysis.api.symbols.KaClassSymbol
 import org.jetbrains.kotlin.analysis.api.symbols.KaFunctionSymbol
-import org.jetbrains.kotlin.analysis.api.types.KaClassType
-import org.jetbrains.kotlin.analysis.api.types.KaFlexibleType
-import org.jetbrains.kotlin.analysis.api.types.KaType
-import org.jetbrains.kotlin.analysis.api.types.KaTypeParameterType
+import org.jetbrains.kotlin.analysis.api.symbols.KaTypeParameterSymbol
+import org.jetbrains.kotlin.analysis.api.types.*
 
 context(KaSession)
 private fun KotlinUsesExtractor.useClassType(
     c: KaClassType
 ): TypeResults {
     // TODO: this cast is unsafe; .symbol is actually a KaClassLikeSymbol
-    val classId = addClassLabel(c.symbol as KaClassSymbol, listOf<Nothing>())
-    val javaResult = TypeResult(classId /* , TODO, TODO */)
+    val javaResult = addClassLabel(c.symbol as KaClassSymbol, c.qualifiers.map { it.typeArguments })
     val kotlinTypeId =
-        tw.getLabelFor<DbKt_class_type>("@\"kt_class;{$classId}\"") {
-            tw.writeKt_class_types(it, classId)
+        tw.getLabelFor<DbKt_class_type>("@\"kt_class;{${javaResult.id}}\"") {
+            tw.writeKt_class_types(it, javaResult.id)
```

Comment:
I don't think this refactoring is going to do the right thing in the future, e.g. for `Int`, where the Java ID might be for `int` but the Kotlin ID should be for `Int`

Reply:
Added a TODO -- I wasn't intending to address this here, only to keep the existing behaviour.